pages/apis/javascript/latest/belt.mdx

```diff
@@ -120,6 +120,55 @@ For example, Belt has the following set modules:
 
 ## Implementation Details
 
+### Array access type safety
+
+One common confusion comes from the way Belt handles array access. It differs from than the default standard library's.
+
+```reason
+let letters = [|"a", "b", "c"|];
+let a = letters[0];
+let capitalA = Js.String2.toUpperCase(a);
+let k = letters[10]; /* Raises an exception! The 10th index doesn't exist. */
+```
+
+Because Belt avoids exceptions and returns `options` instead, this code behaves differently:
+
+```reason
+open Belt;
+let letters = [|"a", "b", "c"|];
+let a = letters[0];
+let captialA = Js.String2.toUpperCase(a); /* Type error! This code will not compile. */
```

Comment:
Maybe it is because I am a newbie but it is not obvious why this is a type error. So the comment could mention that `a` is now of type `option(string)` instead of `string`. What do you think? (I was thinking is type error because it is `String2` instead of `String1`?)

Reply:
I just changed it to `Js.String` for clarity. I couldn’t remember which modules people were using these days, Js.Xxx or Js.Xxx2.

Reply:
@johnridesabike Thanks! my comment was more about the comment could explain why type error happens, like because it is `option(string)` instead of `string`.

Reply:
I don't really know the difference between String modules, but nevermind that I can look up.

Reply:
Oh I see you commented that `a` is `some("a")`. I suppose that should be enough! Thanks!

Reply:
> I couldn’t remember which modules people were using these days, Js.Xxx or Js.Xxx2.

The official statement is here:
https://reasonml.org/apis/javascript/latest/js#jsxxx2-modules

Let's stay away from `xxx2` modules for now in this case!

Reply:
> Oh I see you commented that `a` is `some("a")`. I suppose that should be enough! Thanks!

Thanks, I tried not to over-explain everything in the code comments so it’s not overwhelming. But if anything else isn’t clear just let me know.